src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,85 +16,66 @@ public static class NUnitTestExtractorApp
             {
                 if (parser.ParseArgumentsStrict(args, _options, () => Environment.Exit(-1)))
                 {
                     if (_options.Help)
                     {
                         Console.WriteLine(CommandLine.Text.HelpText.AutoBuild(_options));
+
                         return;
                     }
-                   
-                    if(!string.IsNullOrEmpty(_options.Output))
-                    {
-                        string directory = null;
 
-                        int index = _options.Output.LastIndexOf("/", StringComparison.OrdinalIgnoreCase);
-
-                        if (index > 0)
-                        {
-                            directory = _options.Output.Substring(0, index);
-                        }  
+                    // _options.Output is an abusolutely path of output file for the test name list
+                    if (!string.IsNullOrEmpty(_options.Output))
+                    {
+                        FileInfo fi = new FileInfo(_options.Output);
 
-                        if (directory!=null && !Directory.Exists(directory))
+                        // if the parent directory doesn't exist, then create it
+                        if (!fi.Directory.Exists)
                         {
-                            Directory.CreateDirectory(directory);
+                            fi.Directory.Create();
                         }
 
-                        if (!File.Exists(_options.Output))
+                        // if the output file doesn't exist, then create it
+                        if (!fi.Exists)
                         {
-                            using (FileStream stream = File.Create(_options.Output)) { }
+                            fi.Create();
                         }
                     }
 
                     if (_options.Level != null)
                     {
-                        StreamWriter writer = null;
-                        level = ParseLevel(_options.Level);
-                        try
-                        {
-                            if (_options.Output == null)
-                            {
-                                writer = new StreamWriter(Console.OpenStandardOutput());
-                            }
-                            else
-                            {
-                                writer = new StreamWriter(_options.Output, append: true);
-                            }
-
-                            GetTests(_options.DLLs, level, writer);
-                        }
-                        finally
-                        {
-                            if (writer != null)
-                            {
-                                writer.Dispose();
-                                
-                            }
-                        }
+                        Level level = ParseLevel(_options.Level);
+
+                        GetTests(_options.DLLs, level);
```

Comment:
[Could] The name GetTests isn't quite right since the function doesn't return any tests, it finds them and prints them to the output.

Reply:
That has been changed in the next version.